### PR TITLE
Support for async listeners

### PR DESCRIPTION
**Pull request by iRalph**
Allow async functions as listeners

### CODE REVIEW

**Comment by GrosSacASac** [comment]
please check if it is a function with the `typeof `operator

```
typeof (function x() {})
"function"
typeof (function* x() {})
"function"
typeof (() => {})
"function"
typeof (async () => {})
"function"
typeof (async function x() {})
"function"
```

`if (typeof listener !== 'function') ...`

**Comment by goriunov** [comment]
Actually @GrosSacASac  is right as we are not checking an `object` type it should be alright to use `typeof`.

File: lib/emitter.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ export class EventEmitter {
   private events: { [key: string]: Listener } = {};
 
   public on(event: string, listener: Listener): void {
-    if ({}.toString.call(listener) !== '[object Function]')
+    const s: string = {}.toString.call(listener);
+    if (s !== '[object Function]' && s !== '[object AsyncFunction]')
```
Comment (goriunov) [inline lib/emitter.ts]: Can you change that to `typeof listener === 'function'`

Comment (iRalph) [reply lib/emitter.ts]: Thanks to @GrosSacASac , I've updated the code.

**Pull request merged**